internal/controller/monovertexrollout_controller.go

```diff
@@ -248,10 +252,41 @@ func (r *MonoVertexRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return nil
 }
 
-func mergeMonoVertex(existingMonoVertex *kubernetes.GenericObject, newMonoVertex *kubernetes.GenericObject) *kubernetes.GenericObject {
+func mergeMonoVertex(existingMonoVertex *kubernetes.GenericObject, newMonoVertex *kubernetes.GenericObject) (*kubernetes.GenericObject, error) {
 	resultMonoVertex := existingMonoVertex.DeepCopy()
 	resultMonoVertex.Spec = *newMonoVertex.Spec.DeepCopy()
-	return resultMonoVertex
+	// Use the same replicas as the existing MonoVertex
+	resultMonoVertex, err := withExistingMvtxReplicas(existingMonoVertex, resultMonoVertex)
+	return resultMonoVertex, err
+}
+
+// withExistingMvtxReplicas sets the replicas of the new MonoVertex to the existing MonoVertex's replicas if it exists.
+func withExistingMvtxReplicas(existingMonoVertex, newMonoVertex *kubernetes.GenericObject) (*kubernetes.GenericObject, error) {
+	unstrucExisting, err := kubernetes.ObjectToUnstructured(existingMonoVertex)
+	if err != nil {
+		return newMonoVertex, err
+	}
+	// Have to use float64 as it's  the type of the replicas field in the unstructured object
+	existingReplicas, existing, err := unstructured.NestedFloat64(unstrucExisting.Object, "spec", "replicas")
+	if err != nil {
+		return newMonoVertex, fmt.Errorf("failed to get replicas from existing MonoVertex: %w", err)
+	}
+	if existing {
+		unstrucNew, err := kubernetes.ObjectToUnstructured(newMonoVertex)
+		if err != nil {
+			return newMonoVertex, err
+		}
+		err = unstructured.SetNestedField(unstrucNew.Object, existingReplicas, "spec", "replicas")
+		if err != nil {
+			return newMonoVertex, fmt.Errorf("failed to set replicas in new MonoVertex: %w", err)
+		}
+
+		newMonoVertex, err = kubernetes.UnstructuredToObject(unstrucNew)
+		if err != nil {
+			return newMonoVertex, err
+		}
+	}
+	return newMonoVertex, nil
```

Comment:
Thanks for doing this. Just FYI that Chandan will be working on an [issue](https://github.com/numaproj/numaplane/issues/143) soon to avoid having to do too many of these conversions in the code.